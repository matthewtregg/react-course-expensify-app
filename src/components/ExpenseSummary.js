import React from 'react'
import {connect} from 'react-redux'
import selectExpenses from '../selectors/expenses'
import getTotalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpenseSummary = ({count, total}) => {
const finaltotal = numeral(total/100).format('$0,0.00')

return (
<div>
 {count>1? <p> `viewing {count} expenses totalling {finaltotal}`</p> :
  <p> `viewing {count} expense totalling {finaltotal}`</p> }
</div>
)
}

const mapStateToProps = (state) => {
    
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return ({
       count: visibleExpenses.length,
       total: getTotalExpenses(visibleExpenses)
    })

}






export default connect(mapStateToProps) (ExpenseSummary)