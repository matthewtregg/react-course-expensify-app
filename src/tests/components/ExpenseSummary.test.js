import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary'

test ('should correctly render Expenses summary with one expenses', () => { 
const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={235} />)
expect(wrapper).toMatchSnapshot()
} )

test ('should correctly render Expenses summary with more than one expense', ()=>{ 
const wrapper = shallow(<ExpenseSummary expenseCount={5} expenseTotal={545} />)
expect(wrapper).toMatchSnapshot()

    
} )