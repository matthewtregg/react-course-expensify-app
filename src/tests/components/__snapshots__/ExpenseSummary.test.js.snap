// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses summary with more than one expense 1`] = `
<div>
  <p>
     \`viewing $
     expense totalling $
    $0.00
    \`
  </p>
</div>
`;

exports[`should correctly render Expenses summary with one expenses 1`] = `
<div>
  <p>
     \`viewing $
     expense totalling $
    $0.00
    \`
  </p>
</div>
`;
